### ISSUE TITLE
Ollama does not use gpu acceleration on 7900xtx 

### ISSUE THREAD

**Issue by Redhawk18** [NixOS/nixpkgs#357345]
## Steps To Reproduce

1. I removed all overrides from my 5700xt
2. rebuilt still with `config.rocmSupport = true;`
3. see in btop that my cpu is being pegged

## Expected behavior

For acceleration to just work because this is an officially supported card by amd.

## Additional context
`ollama version is 0.1.38`
Using nixos 24.11 to my knowledge.

## Notify maintainers

@abysssol @dit7ya  @elohmeier 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by azahi** [comment]
Have you tried setting [`services.ollama.rocmOverrideGfx`](https://search.nixos.org/options?channel=unstable&show=services.ollama.rocmOverrideGfx&from=0&size=50&sort=relevance&type=packages&query=ollama.rocmOverrideGfx) to something that your GPU might support? For example `11.0.0`.

**Comment by abysssol** [comment]
> Have you tried setting [`services.ollama.rocmOverrideGfx`](https://search.nixos.org/options?channel=unstable&show=services.ollama.rocmOverrideGfx&from=0&size=50&sort=relevance&type=packages&query=ollama.rocmOverrideGfx) to something that your GPU might support? For example `11.0.0`.

For context, some GPUs that are officially supported don't work without setting `rocmOverrideGfx` (`HSA_OVERRIDE_GFX_VERSION`) ever since #320202 was merged, which reverted a change I made in #312608. The reason it *was* merged, even knowing that, is because otherwise every user would need to compile the rocm dependencies and `ollama-rocm` from scratch (which is so resource intensive that it was causing freezing and crashing on some weaker hardware), due to size limits for what packages are added to the binary cache. It's unfortunate, but there doesn't seem to be a good solution (at least, I wasn't able to find one).
